File: RenzoArredondo/Leccion1/src/system2023mundopc/Raton.java
package system2023mundopc;

public class Raton extends DispositivoEntrada {

	private final int IdRaton;
	private static int contadorRatones;
	
	public Raton(String tipoEntrada,String marca) {
		
		super(tipoEntrada,marca);
		this.IdRaton=++Raton.contadorRatones;
	}

	@Override
	public String toString() {
		return "Raton [IdRaton=" + IdRaton +", "+super.toString()+ "]";
	}
	
	
}

File: RenzoArredondo/Phyton2024/Leccion14/mundo_pc/raton.py

from mundo_pc.dispositivo_entrada import *
class Raton(DispositivoEntrada):

    contador_ratones = 0

    def __init__(self, marca, tipo_entrada):
        Raton.contador_ratones += 1
        self._idRaton = Raton.contador_ratones
        super().__init__(marca, tipo_entrada)


    def __str__(self):
        return f'Id: {self._idRaton}, Marca: {self._marca}, tipo Entrada: {self._tipo_entrada}'



#Hacemos pruebas
if __name__ == "__main__":
    raton1 = Raton('HP', 'USB')
    print(raton1)
    raton2 = Raton('Acer', 'Bluetooth')
    print(raton2)